core/common/src/main/java/alluxio/grpc/GrpcChannelBuilder.java

```diff
@@ -120,18 +119,12 @@ public GrpcChannel build() throws AlluxioStatusException {
     GrpcConnection connection =
         GrpcConnectionPool.INSTANCE.acquireConnection(mChannelKey, mConfiguration);
     try {
-      AuthenticatedChannelClientDriver authDriver = null;
       if (mAuthenticateChannel) {
-        // Create channel authenticator based on provided content.
-        ChannelAuthenticator channelAuthenticator = new ChannelAuthenticator(
-            connection, mParentSubject, mAuthType, mConfiguration);
-        // Authenticate a new logical channel.
-        channelAuthenticator.authenticate();
-        // Acquire authentication driver.
-        authDriver = channelAuthenticator.getAuthenticationDriver();
+        return new GrpcChannel(connection,
+            AuthenticationUtils.authenticate(connection, mParentSubject, mAuthType));
       }
       // Return a wrapper over logical channel.
-      return new GrpcChannel(connection, authDriver);
+      return new GrpcChannel(connection, null);
```

Comment:
Shall we make authDrive as Optional in the GrpcChanel's constructor?  it's might be out of scope of this PR, your call

Reply:
Ya, i have a later PR that would merge `GrpcChannel` and `GrpcConnection`, as well as `ChannelKey` and `ConnectionKey`, which will also fix this. Getting the easier stuff out of the way first.